src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -822,6 +832,7 @@ void handleConnack(MqttConnAckMessage msg) {
       if (this.connectHandler != null) {
 
         if (msg.code() == MqttConnectReturnCode.CONNECTION_ACCEPTED) {
+          this.isConnected = true;
```

Comment:
As the user might choose to provide `null` as `connectHandler`doing the assignment inside the `if (this.connectHandler != null)` would result in `isConnected()` to return `false` in such cases.

I think the assignment should be re-written in a way that it takes care of this, like:

~~~java
this.isConnected = msg.code() = MqttConnectReturnCode.CONNECTION_ACCEPTED;

if (this.connectHandler != null) {
…
~~~

Reply:
You are absolutely right !

Reply:
You are absolutely right !